tests/integration/article.test.js

```diff
@@ -224,60 +225,330 @@ describe("# Articles endpoints", () => {
+        .end((err, res) => {
+          expect(res.status).equals(OK);
+          expect(res.body.message).to.contain("Successful");
+          expect(res.body)
+            .to.haveOwnProperty("articles")
+            .to.be.an("array");
+          expect(res.body)
+            .to.haveOwnProperty("articlesCount")
+            .to.be.a("number");
+          res.body.articles.map(article => {
+            expect(article).to.have.any.keys(
+              "slug",
+              "title",
+              "description",
+              "body",
+              "tagsList",
+              "favorited",
+              "favoritesCount",
+              "author"
+            );
+            expect(article.tagsList).to.be.an("array");
+            expect(article.favoritesCount).to.be.a("number");
+            expect(article.favorited).to.be.a("boolean");
+            return expect(article.author)
+              .to.be.an("object")
+              .to.haveOwnProperty("username")
+              .to.be.a("string");
+          });
+          done();
+        });
+    });
+
+    it("should return a list of all created articles even when no token provided", done => {
+      chai
+        .request(app)
+        .get(`/api/v1/articles`)
+        .end((err, res) => {
+          expect(res.status).equals(OK);
+          expect(res.body.message).to.contain("Successful");
+          expect(res.body)
+            .to.haveOwnProperty("articles")
+            .to.be.an("array");
+          expect(res.body)
+            .to.haveOwnProperty("articlesCount")
+            .to.be.a("number");
+
+          res.body.articles.map(article => {
+            expect(article).to.have.any.keys(
+              "slug",
+              "title",
+              "description",
+              "body",
+              "tagsList",
+              "favorited",
+              "favoritesCount",
+              "author"
+            );
+            expect(article.tagsList).to.be.an("array");
+            expect(article.favoritesCount).to.be.a("number");
+            expect(article.favorited).to.be.a("boolean");
+            return expect(article.author)
+              .to.be.an("object")
+              .to.haveOwnProperty("username")
+              .to.be.a("string");
+          });
+          done();
+        });
+    });
+
+    describe("Pagination (limit the number of articles to be returned)", () => {
+      it("should return the exact number of articles specified by the limit param", done => {
+        chai
+          .request(app)
+          .get(`/api/v1/articles?limit=2`)
+          .end((err, res) => {
+            expect(res.status).equals(OK);
+            expect(res.body.message).to.contain("Successful");
+            expect(res.body)
+              .to.haveOwnProperty("articles")
+              .to.be.an("array");
+            expect(res.body)
+              .to.haveOwnProperty("articlesCount")
+              .to.be.a("number");
+            expect(res.body.articles.length).to.equal(2);
```

Comment:
Why these 2 assertions?
Isn't it redudant?
Can you please keep one assertion?

Reply:
Thanks for raising such a good idea, but I'm afraid I don't agree with you. Now that this feature is implementing the pagination functionality, I'm fetching the first page containing the first x number of data and raise another request to get the next page's data and finally compare them to make sure they contain different data as expected.